icd/api/pipeline_compiler.cpp

```diff
@@ -1047,24 +1047,89 @@ void PipelineCompiler::GetPipelineCreationInfoNext(
+            {
+                stageCreationFeedbacks[i].flags = 0;
+            }
+        }
 
+        if ((pPipelineCreationFeadbackCreateInfo->pPipelineStageCreationFeedbacks != nullptr) &&
+            (pStageFeedback != nullptr))
+        {
+            // Compute pipeline does have VkGraphicsPipelineCreateInfo.
+            auto *stageCreationFeedbacks = pPipelineCreationFeadbackCreateInfo->pPipelineStageCreationFeedbacks;
+            if (pCreateInfo == nullptr)
+            {
+                stageCreationFeedbacks[0].flags |= VK_PIPELINE_CREATION_FEEDBACK_VALID_BIT_EXT;
+                if (pStageFeedback[0].hitApplicationCache)
+                {
+                    stageCreationFeedbacks[0].flags |=
+                        VK_PIPELINE_CREATION_FEEDBACK_APPLICATION_PIPELINE_CACHE_HIT_BIT_EXT;
+                }
+                stageCreationFeedbacks[0].duration = pStageFeedback[0].duration;
+            }
+            else
+            {
+                VK_ASSERT(pCreateInfo->stageCount <= ShaderStage::ShaderStageGfxCount);
+                for (uint32_t i = 0; i < pCreateInfo->stageCount; ++i)
+                {
+                    uint32_t feedbackStage = 0;
+                    VK_ASSERT(pCreateInfo->pStages[i].sType ==
+                              VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO);
+                    switch (pCreateInfo->pStages[i].stage)
+                    {
+                        case VK_SHADER_STAGE_VERTEX_BIT:
+                            feedbackStage = ShaderStage::ShaderStageVertex;
+                            break;
+                        case VK_SHADER_STAGE_TESSELLATION_CONTROL_BIT:
+                            feedbackStage = ShaderStage::ShaderStageTessControl;
+                            break;
+                        case VK_SHADER_STAGE_TESSELLATION_EVALUATION_BIT:
+                            feedbackStage = ShaderStage::ShaderStageTessEval;
+                            break;
+                        case VK_SHADER_STAGE_GEOMETRY_BIT:
+                            feedbackStage = ShaderStage::ShaderStageGeometry;
+                            break;
+                        case VK_SHADER_STAGE_FRAGMENT_BIT:
+                            feedbackStage = ShaderStage::ShaderStageFragment;
+                            break;
+                        default:
+                            VK_NEVER_CALLED();
+                            break;
+                    }
+                    stageCreationFeedbacks[i].flags |= VK_PIPELINE_CREATION_FEEDBACK_VALID_BIT_EXT;
+                    if (pStageFeedback[feedbackStage].hitApplicationCache)
+                    {
+                        stageCreationFeedbacks[i].flags |=
+                            VK_PIPELINE_CREATION_FEEDBACK_APPLICATION_PIPELINE_CACHE_HIT_BIT_EXT;
+                    }
+                    stageCreationFeedbacks[i].duration = pStageFeedback[feedbackStage].duration;
```

Comment:
I think you should create a helper function that converts a PipelineCreationFeedback to a VkPipelineCreationFeedbackEXT and use for both the pipeline and per stage information.  The zeroing of the flags can be done there.

Reply:
done